src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -319,6 +319,28 @@
       (middleware-utils/json-response 200 info-for-json))))
 
 (schema/defn ^:always-validate
+  all-tasks-response! :- ringutils/RingResponse
+  "Process the info, returning a Ring response to be propagated back up to the
+  caller of the endpoint.
+  Returns environment as a list of objects for an eventual future in which
+  tasks for all environments can be requested, and a given task will list all
+  the environments it is found in."
+  [info-from-jruby :- List
```

Comment:
Yeah, at the time I wasn't super sure how to write it because JRuby, but then that turned out to not be a thing for this function's input. I can make this better.